apps/api/src/domain/models/resource.entity.ts

```diff
@@ -13,20 +13,31 @@ import { Aggregate } from './aggregate.entity';
 import { getAllowedContextsForModel } from './allowedContexts/isContextAllowedForGivenResourceType';
 import { EdgeConnectionContext } from './context/context.entity';
 import { EdgeConnectionContextType } from './context/types/EdgeConnectionContextType';
+import { AccessControlList } from './shared/access-control/access-control-list.entity';
 
 export abstract class Resource extends Aggregate implements HasAggregateId {
     readonly type: ResourceType;
 
     // TODO: Rename this 'isPublished' - db migration
     readonly published: boolean;
 
+    /**
+     * TODO We need a migration to make the acl required. We'll also need to
+     * populate this on all test data.
+     */
+    readonly queryAccessControlList?: AccessControlList;
+
     constructor(dto: DTO<Resource>) {
         super(dto);
 
         // This should only happen in the validation flow
         if (!dto) return;
 
-        this.published = typeof dto.published === 'boolean' ? dto.published : false;
+        const { published, queryAccessControlList: aclDto } = dto;
```

Comment:
Just wondering why acl is not spelled out in aclDto?  Is that a regular practice?

Reply:
Note that I spell it out in the prop `queryAccessControlList`, but here I feel it would be tedious. I guess if I wanted to really spell things out, it would be `accessControlListDataTransferObject`. `ACL` is almost as common an acronym as `DTO`.